jmx-scraper/src/integrationTest/java/io/opentelemetry/contrib/jmxscraper/target_systems/MetricsVerifier.java

```diff
@@ -63,6 +63,9 @@ public MetricsVerifier disableStrictMode() {
    */
   @CanIgnoreReturnValue
   public MetricsVerifier add(String metricName, Consumer<MetricAssert> assertion) {
+    if (assertions.containsKey(metricName)) {
```

Comment:
[for reviewer] this ensures that we don't have duplicated assertions for the same metric.